pkg/controller/installation/core_controller.go

```diff
@@ -1446,6 +1448,7 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 	} else {
 		instance.Status.ImageSet = imageSet.Name
 	}
+	instance.Status.CalicoVersion = calicoVersion
```

Comment:
I'd like to see this line right after the `instance.Status.Variant` assignment and a comment that says the two fields should be updated at the same time so there isn't a chance of an inconsistent status state.